test/controllers/callback_controller_test.rb

```diff
@@ -296,20 +289,40 @@ class CallbackControllerTest < ActionController::TestCase
         config.webhooks = [{ topic: "carts/update", address: "example-app.com/webhooks" }]
       end
 
-      ShopifyApp::WebhooksManager.expects(:queue).with("shop", "token")
+      ShopifyApp::WebhooksManager.expects(:queue).with(SHOP_DOMAIN, "token")
 
       get :callback, params: @callback_params
       assert_response 302
     end
 
+    test "#callback performs install_webhook job with an offline session after an online session OAuth" do
+      shop_storage = ShopifyApp::InMemoryShopSessionStore
+
+      ShopifyApp.configure do |config|
+        config.webhooks = [{ topic: "carts/update", address: "example-app.com/webhooks" }]
+        config.shop_session_repository = shop_storage
+        config.user_session_repository = ShopifyApp::InMemoryUserSessionStore
+      end
+      mock_oauth(session: online_session)
+      shop_storage.store(@stubbed_session)
+
+      ShopifyApp::WebhooksManager.expects(:queue).with(SHOP_DOMAIN, "token")
+
+      get :callback, params: @callback_params
+      assert_response 302
+    rescue => e
+      shop_storage.clear
+      raise e
+    end
```

Comment:
i think using `ensure` is more suitable in this usecase

```suggestion
    ensure
      shop_storage.clear
    end
```